### ISSUE TITLE
Collision begin_handler/pre_handler should return whether the collision should be processed or ignored

### ISSUE THREAD

**Issue by WebF0x** [pythonarcade/arcade#927]
### Actual behavior:

When a collision begin_handler/pre_handler returns False, the collision is NOT ignored

### Expected behavior:

When a collision begin_handler/pre_handler returns False, the collision is ignored

### Steps to reproduce/example code:

Add a collision begin_handler/pre_handler that returns False

```
def ignore_collision_handler(_, _, _, _, _):
    return False

self.physics_engine.add_collision_handler(
    "thing",
    "thing",
    begin_handler=ignore_collision_handler
)
```

When the "things" collide, the collision will not be ignored and you will get this warning:
```
[...] arcade\pymunk_physics_engine.py:258: UserWarning: Function '_f1' should return a bool to indicate if the collision should be processed or not when used as 'begin' or 'pre_solve' collision callback.
  def _f1(arbiter, space, data):
```

### What should be added/changed?

According to PyMunk's doc, the begin_handler should "Return true from the callback to process the collision normally or false to cause pymunk to ignore the collision entirely."
http://www.pymunk.org/en/latest/pymunk.html#pymunk.CollisionHandler.begin

begin_handler and pre_handler should return whether the collision should be processed

In pymunk_physics_engine.py,

```
        def _f1(arbiter, space, data):
            sprite_a, sprite_b = self.get_sprites_from_arbiter(arbiter)
            begin_handler(sprite_a, sprite_b, arbiter, space, data)
```

the function _f1 should return what is returned by begin_handler like this:
```
        def _f1(arbiter, space, data):
            sprite_a, sprite_b = self.get_sprites_from_arbiter(arbiter)
            # Change below!
            # Probably cannot return directly for backwards compatibility.
            # This used to return None, and the collisions were processed. So I guess we can't assume None is False.
            should_process_collision = begin_handler(sprite_a, sprite_b, arbiter, space, data) != False
            return should_process_collision
```

### What would it help with?

Allow for collisions of a certain type to be ignored

**Comment by pvcraven** [comment]
Thanks for the report. Fix should be in 2.6 release.

**Comment by Cleptomania** [comment]
Released with 2.6.0
